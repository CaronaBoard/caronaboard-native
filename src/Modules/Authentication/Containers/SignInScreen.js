import Styles from './Styles/SignInScreenStyles'
import React, { Component } from 'react'
import { View } from 'react-native'
import { RkText } from 'react-native-ui-kitten'
import { Button, TextInput } from '../../Shared/Components'
import { signInFirebase } from '../../../Redux/Actions'
import { connect } from 'react-redux'

export class SignInScreen extends Component {
  constructor (props) {
    super()
    this.state = {
      email: '',
      password: '',
      loading: false
    }
  }

  onButtonSubmit = () => {
    this.props.signIn(this.state.email, this.state.password).then(this.onLoginSuccess)
  }

  onLoginSuccess = (userData) => {
    // TODO navigate to ridelist?
  }

  render () {
    return (
      <View style={Styles.flexible}>
        <View style={Styles.container}>
          <RkText style={Styles.title}>
            SIGN IN
          </RkText>
          <View style={Styles.credentialsContainer}>
            <TextInput label='EMAIL ADDRESS' onChangeText={(email) => { this.setState({email}) }} />
            <TextInput label='PASSWORD' onChangeText={(password) => { this.setState({password}) }} />
          </View>
          <Button text='SIGN IN' onPress={this.onButtonSubmit} />
        </View>
      </View>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    userData: state.auth.userData
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    signIn: (e, p) => dispatch(signInFirebase(e, p))
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(SignInScreen)
