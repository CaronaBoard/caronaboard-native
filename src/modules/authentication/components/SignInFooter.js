import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import { RkButton, RkText } from 'react-native-ui-kitten'
import Styles from './styles/SignInFooterStyles'
import { screens } from '../../../configuration/navigation/Screens'

export class SignInFooter extends Component {
  static propTypes = {
    email: PropTypes.string,
    navigator: PropTypes.object.isRequired
  }

  navigateTo = screen => () => this.props.navigator.push({screen})

  render () {
    return (
      <View>
        <View style={Styles.textRow}>
          <RkButton
            key='forgot-password-button'
            rkType='clear'
            onPress={this.navigateTo(screens.forgotPassword.id)}
          >
            <RkText rkType='header6'>Forgot your password?</RkText>
          </RkButton>
        </View>
        <View style={Styles.textRow}>
          <RkText rkType='primary3'>Donâ€™t have an account?</RkText>
          <RkButton
            key='signup-button'
            rkType='clear'
            onPress={this.navigateTo(screens.signUp.id)}
          >
            <RkText rkType='header6'> Sign up now</RkText>
          </RkButton>
        </View>
      </View>
    )
  }
}
