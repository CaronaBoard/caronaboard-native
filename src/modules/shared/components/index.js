export { Button } from './Button'
export { TextInput } from './TextInput'
export { GradientButton } from './GradientButton'
export { LinearGradient } from './LinearGradient'
