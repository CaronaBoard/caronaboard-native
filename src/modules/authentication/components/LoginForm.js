import React from 'react'
import { View, Image, Keyboard } from 'react-native'
import Toast, { DURATION } from 'react-native-easy-toast'
import { RkText, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { GradientButton, TextInput } from '../../shared/components'
import { styles } from '../containers/styles/LoginScreenStyles'

export class LoginForm extends React.Component {
  static defaultProps = {
    footer: () => {},
    toast: {
      showAlert: false,
      message: ''
    }
  }

  state = {
    email: '',
    password: '',
    loading: false
  }

  onButtonPress = () => {
    const { email, password } = this.state
    this.props.onButtonPress(email, password)
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.toast

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
    }
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/caronaboard-logo-azul.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <TextInput
          onChangeText={(email) => { this.setState({email}) }}
          key='email-input'
          rkType='rounded'
          placeholder='Username'
          autoCapitalize='none'
        />
        <TextInput
          onChangeText={(password) => { this.setState({password}) }}
          key='password-input'
          rkType='rounded'
          placeholder='Password'
          secureTextEntry
        />

        <View style={styles.centralized}>
          <GradientButton
            rkType='large'
            style={styles.loginButton}
            text={this.props.buttonText}
            onPress={this.onButtonPress}
          />
        </View>

        {this.props.footer(this.state.email)}
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}
