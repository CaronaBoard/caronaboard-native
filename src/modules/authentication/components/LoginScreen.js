import React from 'react'
import { View, Image, Keyboard } from 'react-native'
import Toast, { DURATION } from 'react-native-easy-toast'
import { RkText, RkTextInput, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { GradientButton } from '../../shared/components/index'
import { styles } from '../containers/styles/LoginScreenStyles'

export const INITIAL_STATE = {
  email: '',
  password: '',
  loading: false
}

export default class LoginV2 extends React.Component {
  constructor (props) {
    super(props)
    this.state = INITIAL_STATE
  }

  onButtonPress = () => {
    const { email, password } = this.state
    const { onButtonPress } = this.props
    try {
      onButtonPress(email, password)
    } catch (error) {
      console.log(error)
    }
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.alert

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
    }
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/logo.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <RkTextInput onChangeText={(email) => { this.setState({email}) }} rkType='rounded' placeholder='Username' autoCapitalize='none' />
        <RkTextInput onChangeText={(password) => { this.setState({password}) }} rkType='rounded' placeholder='Password' secureTextEntry />

        <View style={styles.centralized}>
          <GradientButton
            style={styles.loginButton}
            rkType='large'
            text='LOGIN'
            onPress={this.onButtonPress}
          />
        </View>

        {this.props.footer()}
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}
