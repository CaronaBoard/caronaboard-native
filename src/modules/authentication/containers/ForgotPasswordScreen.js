import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { forgotPasswordFirebase } from '../../../redux/actions'
import { RkText, RkTextInput, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { styles } from '../containers/styles/LoginScreenStyles'
import { View, Image, Keyboard } from 'react-native'
import { GradientButton } from '../../shared/components/index'
import Toast, { DURATION } from 'react-native-easy-toast'

export class ForgotPasswordScreen extends Component {
  static defaultProps = {
    toast: {
      showAlert: false,
      message: ''
    }
  }

  static propTypes = {
    resetPassword: PropTypes.func.isRequired,
    alert: PropTypes.shape({
      showAlert: PropTypes.bool.isRequired,
      message: PropTypes.string
    }).isRequired
  }

  state = {
    email: '',
    alert: {
      showAlert: false,
      message: ''
    }
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.alert

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
      setTimeout(() => {
        this.props.navigator.push({
          screen: 'carona.signIn',
          title: 'Sign In Screen'
        })
      }, DURATION.LENGTH_LONG)
    }
  }

  render () {
    const { resetPassword } = this.props
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/caronaboard-logo-azul.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <RkTextInput
          onChangeText={(email) => { this.setState({email}) }}
          key='email-input'
          rkType='rounded'
          placeholder='Username'
          autoCapitalize='none'
        />
        <View style={styles.centralized}>
          <GradientButton
            rkType='large'
            style={styles.loginButton}
            text='Send'
            onPress={() => resetPassword(this.state.email)}
          />
        </View>
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    alert: state.auth.alert
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    resetPassword: (email) => dispatch(forgotPasswordFirebase(email))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ForgotPasswordScreen)
