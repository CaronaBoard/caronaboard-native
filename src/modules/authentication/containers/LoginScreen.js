import React from 'react'
import {
  View,
  Image,
  Keyboard
} from 'react-native'
import {
  RkButton,
  RkText,
  RkTextInput,
  RkAvoidKeyboard,
  RkTheme
} from 'react-native-ui-kitten'
import { GradientButton } from '../../shared/components'
import { styles } from './styles/LoginScreenStyles'

export default class LoginV2 extends React.Component {
  onLoginButtonPress = () => {
  }

  navigateToSignUpScreen = () => {
  }

  renderIcon = () => {
    let source
    if (RkTheme.current.name === 'light') {
      source = require('../../../assets/images/logo.png')
    } else {
      source = require('../../../assets/images/logoDark.png')
    }

    return <Image style={styles.image} source={source} />
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          {this.renderIcon()}
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <RkTextInput rkType='rounded' placeholder='Username' />
        <RkTextInput rkType='rounded' placeholder='Password' secureTextEntry />

        <View style={styles.centralized}>
          <GradientButton style={styles.loginButton} rkType='large' text='LOGIN' onPress={this.onLoginButtonPress} />
        </View>

        <View style={styles.textRow}>
          <RkText rkType='primary3'>Donâ€™t have an account?</RkText>
          <RkButton rkType='clear' onPress={this.navigateToSignUpScreen}>
            <RkText rkType='header6'> Sign up now </RkText>
          </RkButton>
        </View>

      </RkAvoidKeyboard>
    )
  }
}
