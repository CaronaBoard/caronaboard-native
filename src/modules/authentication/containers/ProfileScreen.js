import Styles from './styles/ProfileScreenStyles'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { View, TouchableOpacity } from 'react-native'
import { RkText, RkChoice, RkChoiceGroup } from 'react-native-ui-kitten'
import type { ProfileType } from '../../../services/firebase/types'
import { ProfilePropType } from '../types'

import { TextInput, GradientButton } from '../../shared/components'
import { saveProfileFirebase } from '../../../redux/actions'

export const CONTACT_OPTIONS = ['Whatsapp', 'Telegram']
export const INITIAL_STATE: ProfileType = {
  profile: {
    name: '',
    uid: '',
    contact: {
      kind: '',
      value: ''
    }
  },
  loading: false
}

export class ProfileScreen extends Component {
  static propTypes = {
    navigator: PropTypes.object.isRequired,
    userId: PropTypes.string.isRequired,
    profile: ProfilePropType
  }

  state = INITIAL_STATE

  onContactOptionSelected = kind => {
    const { profile } = this.state
    let { contact } = profile
    contact.kind = kind
    this.setState({profile: {...profile, contact}})
  }

  onButtonSubmit = async () => {
    this.setState({loading: true})

    try {
      this.props.saveProfile(this.state.profile)
    } catch (error) {
      console.error(error)
    }

    this.setState({loading: false})
  }

  renderOption = (option, i) => (
    <View key={i}>
      <TouchableOpacity choiceTrigger>
        <View style={Styles.contactOption}>
          <RkChoice rkType='radio' />
          <RkText>{option}</RkText>
        </View>
      </TouchableOpacity>
    </View>
  )

  componentDidMount () {
    const { profile } = this.props
    this.setState({profile, loading: false})
  }

  render () {
    const { profile } = this.state
    const { contact, name } = profile

    return (
      <View style={Styles.flexible}>
        <View style={Styles.container}>
          <RkText style={Styles.title}>
            PROFILE
          </RkText>
          <View style={Styles.inputTextsContainer}>
            <TextInput
              placeholder='Your name'
              value={name}
              onChangeText={name => { this.setState({profile: {...profile, name}}) }}
            />
          </View>
          <RkChoiceGroup
            radio
            rkType='clear'
            onChange={index => this.onContactOptionSelected(CONTACT_OPTIONS[index])}>
            {CONTACT_OPTIONS.map(this.renderOption)}
          </RkChoiceGroup>
          <View style={Styles.inputTextsContainer}>
            <TextInput
              placeholder='Number'
              value={contact.value}
              onChangeText={value => { this.setState({profile: {...profile, contact: {...contact, value}}}) }}
            />
          </View>
        </View>
        <View style={Styles.centralized}>
          <GradientButton
            rkType='large'
            text={'SAVE'}
            onPress={this.onButtonSubmit}
          />
        </View>
      </View>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    profile: state.auth.profile, // TODO: do we have profile saved into state?
    userId: state.auth.userData.uid
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    saveProfile: (profile, userId) => dispatch(saveProfileFirebase(profile, userId))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProfileScreen)
