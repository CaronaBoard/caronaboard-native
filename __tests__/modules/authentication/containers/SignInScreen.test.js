import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { SignInScreen, INITIAL_STATE } from '../../../../src/modules/authentication/containers/SignInScreen'
import { Button } from '../../../../src/modules/shared/components'

const props = {
  userData: {},
  signIn: jest.fn()
}

describe('<SignInScreen />', () => {
  const email = 'duduzinho@uol.com'
  const password = 'marotagem'
  const wrapper = shallow(<SignInScreen {...props} />)

  beforeEach(() => {
    wrapper.setState(INITIAL_STATE)
  })

  it('Should have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should update SignInScreen state as text input changes', () => {
    const emailInput = wrapper.findWhere(node => node.key() === 'email-input')
    const passwordInput = wrapper.findWhere(node => node.key() === 'password-input')

    // Find out a better way to simulate onChangeText
    emailInput.prop('onChangeText')(email)
    passwordInput.prop('onChangeText')(password)

    expect(wrapper.state('email')).toEqual(email)
    expect(wrapper.state('password')).toEqual(password)
  })

  it('Should call signIn passing username and password', () => {
    const submitButton = wrapper.find(Button)

    wrapper.setState({email, password})
    submitButton.simulate('press')

    expect(props.signIn).toHaveBeenCalledWith(email, password)
  })

  it.skip('Should redirect if user is already loggedIn', () => {
    // TODO:
  })
})
