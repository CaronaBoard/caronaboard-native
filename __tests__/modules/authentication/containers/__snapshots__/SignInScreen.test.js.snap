// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInScreen /> Should have a snapshot 1`] = `
<LoginForm
  buttonText="SIGN IN"
  footer={[Function]}
  onButtonPress={[Function]}
  toast={
    Object {
      "message": "",
      "showAlert": false,
    }
  }
/>
`;
